astra/src/test/java/com/slack/astra/zipkinApi/ZipkinServiceTest.java

```diff
@@ -113,14 +166,354 @@ public void testGetTraceByTraceId_respectsMaxSpans() throws Exception {
+
+      verify(searcher)
+          .doSearch(
+              Mockito.argThat(
+                  request ->
+                      request.getHowMany() == maxSpansParam
+                          && request.getQuery().contains("\"trace_id\":\"" + traceId + "\"")));
+
+      verify(mockBlobStore).upload(Mockito.anyString(), Mockito.any(Path.class));
+
+      Path directoryDownloaded = Files.createTempDirectory(traceId);
+      mockBlobStore.download(
+          String.format("%s/%s", TRACE_CACHE_PREFIX, traceId), directoryDownloaded);
+
+      File[] filesDownloaded = directoryDownloaded.toFile().listFiles();
+      assertThat(Objects.requireNonNull(filesDownloaded).length).isEqualTo(1);
+      Path uploadedFile = filesDownloaded[0].toPath();
+      assertThat(uploadedFile.toString()).endsWith("traceData.json.gz");
+      String data = ZipkinService.decompressJsonData(Files.readAllBytes(uploadedFile));
+      assertNotNull(data, "Decompressed data should not be null");
+    }
+  }
+
+  @Test
+  public void testGetTraceByTraceId_respectUserRequest_respectDataFreshness_skip_search()
+      throws Exception {
+    try (MockedStatic<Tracing> mockedTracing = mockStatic(Tracing.class)) {
+      // Mocking Tracing and Span
+      Tracer mockTracer = mock(Tracer.class);
+      Span mockSpan = mock(Span.class);
+
+      mockedTracing.when(Tracing::currentTracer).thenReturn(mockTracer);
+      when(mockTracer.currentSpan()).thenReturn(mockSpan);
+
+      String traceId = "test_trace_4";
+
+      Path filePath = Paths.get(Resources.getResource("zipkinApi/traceData.json").toURI());
+
+      byte[] data = ZipkinService.compressJsonData(Files.readString(filePath));
+      Path tempDir = Files.createTempDirectory(traceId);
+      Path traceFile = tempDir.resolve("traceData.json.gz");
+      Files.write(traceFile, data);
+
+      String path = String.format("%s/%s", TRACE_CACHE_PREFIX, traceId);
+      mockBlobStore.upload(path, tempDir);
+
+      int maxSpansParam = 10000;
+      boolean userRequest = true;
+      long dataFreshnessInMinutes = 0;
+
+      zipkinService.getTraceByTraceId(
+          traceId,
+          Optional.empty(),
+          Optional.empty(),
+          Optional.of(maxSpansParam),
+          Optional.of(userRequest),
+          Optional.of(dataFreshnessInMinutes));
+
+      verify(mockBlobStore).download(Mockito.anyString(), Mockito.any(Path.class));
+
+      Path directoryDownloaded = Files.createTempDirectory(traceId);
+      mockBlobStore.download(
+          String.format("%s/%s", TRACE_CACHE_PREFIX, traceId), directoryDownloaded);
+
+      File[] filesDownloaded = directoryDownloaded.toFile().listFiles();
+      assertThat(Objects.requireNonNull(filesDownloaded).length).isEqualTo(2);
+      for (File file : filesDownloaded) {
+        if (file.getName().endsWith("traceData.json.gz")) {
+          Path uploadedFile = file.toPath();
+          assertThat(uploadedFile.toString()).endsWith("traceData.json.gz");
```

Comment:
I think either drop the if condition for `traceData.json.gz` or drop the assertion, since they appear to be checking the same thing?